### PR TITLE
Fix/sp42 support author affiliation dev37

### PR DESCRIPTION
**Pull request by weko3-dev22**

### CODE REVIEW

File: modules/weko-schema-ui/weko_schema_ui/schema.py [hunk 1]
```diff
@@ -246,13 +246,18 @@ def __init__(self, record=None, schema_name=None):
     def get_ignore_item_from_option(self):
         """Get all keys of properties that is enable Hide option in metadata."""
         ignore_list = []
+        ignore_list_all = []
+        ignore_list_all_bk = {}
         from weko_records.utils import get_options_and_order_list
```
Comment (weko3-dev22) [inline modules/weko-schema-ui/weko_schema_ui/schema.py:251]: Move to top file.

File: modules/weko-deposit/weko_deposit/api.py [hunk 2]
```diff
@@ -1346,8 +1346,44 @@ def _get_creator_to_show_popup(self, creators: Union[list, dict],
...
+                                        affiliation_names[i])
+                                    self._get_creator_to_show_popup(
+                                        [affiliation_name_identifiers[i]],
+                                        language, creator_list,
+                                        creator_list_temp)
+                                else:
+                                    self._get_creator_to_show_popup(
+                                        [affiliation_name_identifiers[i]],
+                                        language, creator_list,
+                                        creator_list_temp)
```
Comment (ngvanbi) [inline modules/weko-deposit/weko_deposit/api.py:1382]: Create a new method for this source code.

Example:
```python
def _abc(_names_1, _names_2, language, creator_list, creator_list_temp):
    for i in range(len(_names_1)):
        if i < len(_names_2):
            _names_1[i].update(
                _names_2[i])
            self._get_creator_to_show_popup(
                [_names_1[i]],
                language, creator_list,
                creator_list_temp)
        else:
            self._get_creator_to_show_popup(
                [_names_1[i]],
                language, creator_list,
                creator_list_temp)
            

def format_affiliation(value):
    for create in value:
        affiliation_names = create.get('affiliationNames')
        affiliation_name_identifiers = create.get('affiliationNameIdentifiers')
        _names_1 = None
        _names_2 = None
        if len(affiliation_names) >= len(
                affiliation_name_identifiers):
            _names_1 = affiliation_names
            _names_2 = affiliation_name_identifiers
        else:
            _names_1 = affiliation_name_identifiers
            _names_2 = affiliation_names

        _abc(_names_1, _names_2, language, creator_list, creator_list_temp)

```

File: modules/weko-records/weko_records/utils.py [hunk 3]
```diff
@@ -482,53 +488,70 @@ def get_values(v):
...
+                            s['parent_option'] = {
+                                'required': option.get("required"),
+                                'show_list': option.get("showlist"),
+                                'specify_newline': option.get("crtf"),
+                                'hide': option.get("hidden")
+                            }
+                            break
+            # else:
+            #     data = val.get('attribute_value')
+            #     lst[-1] = data
```
Comment (ngvanbi) [inline modules/weko-records/weko_records/utils.py:536]: Remove this comment code

File: modules/weko-records/weko_records/utils.py [hunk 4]
```diff
@@ -482,53 +488,70 @@ def get_values(v):
...
+                is_specify_newline = parent_option.get(
+                    'specify_newline') if parent_option.get(
+                    'specify_newline') else option.get('specify_newline')
+                is_hide = parent_option.get('hide') if parent_option.get(
+                    'hide') else option.get('hide')
+                if not is_hide and is_show_list:
+                    if is_specify_newline or len(items) == 0:
+                        items.append(value)
+                    else:
+                        items[-1] += "," + value
```
Comment (ngvanbi) [inline modules/weko-records/weko_records/utils.py:554]: Create a new method for this source code

File: modules/weko-records/weko_records/utils.py [hunk 5]
```diff
@@ -626,16 +649,20 @@ def to_sort_dict(alst, klst):
                     for lst in klst:
                         key = lst[0].split('.')[-1]
                         val = alst.pop(key, {})
+                        hide = lst[3]['hide']
```
Comment (ngvanbi) [inline modules/weko-records/weko_records/utils.py:652]: Check if `lst[3]` has a hidden property?

**Pull request merged**